generator/internal/metadata_decorator_generator.cc

```diff
@@ -406,6 +406,11 @@ void $metadata_class_name$::SetMetadata(grpc::ClientContext& context,
   }
   auto const& authority = options.get<AuthorityOption>();
   if (!authority.empty()) context.set_authority(authority);
+  if (options.has<google::cloud::CustomHeadersOption>()) {
+    for (auto const& h : options.get<google::cloud::CustomHeadersOption>()) {
```

Comment:
Drop the `options.has<>` conditional?

Also, s/google::cloud::CustomHeadersOption/CustomHeadersOption/